### PR TITLE
Git pull multi process

### PR DESCRIPTION
**Pull request by mrubiosan**
This speeds up the pulling process by forking up to 4 processes executing in parallel.

### CODE REVIEW

**Comment by sanglt** [comment]
Hi @mrubiosan:

This doesn't help the pull image faster. Because after download a layer, docker daemon need to unarchive its, and this process doesn't support running in multi process yet.

To make its simple, you can update param  in your docker client to test:
max-concurrent-downloads,
max-concurrent-uploads


https://docs.docker.com/engine/reference/commandline/dockerd/

**Comment by sanglt** [comment]
You right. Sorry about my comment above.

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ GO1 monolith
 
     php scripts/stop.php
     php scripts/pull.php
-    php scripts/git/pull.php --confirm # Add --reset if you want to reset the code
+    php scripts/git/pull.php --confirm# Add --reset if you want to reset the code, --faster to spawn multiple processes
```
Comment (tienvx) [inline README.md]: I think we need to keep the space to prevent typo when someone copy paste this command.

File: scripts/git/pull.php [hunk 2]
```diff
@@ -17,7 +18,10 @@
 
 $defaultBranch = 'master';
 
-return call_user_func(function () use ($confirm, $reset, $pwd, $projects, $projectsMap, $custom, $defaultBranch) {
+return call_user_func(function () use ($confirm, $reset, $pwd, $projects, $projectsMap, $custom, $defaultBranch, $multiProcess) {
+    $canFork = $multiProcess && function_exists('pcntl_fork');
+    $maxProcesses = 4;
+    $children = [];
```
Comment (andytruong) [inline scripts/git/pull.php]: children -> processes?

**Pull request merged**